### ISSUE TITLE
第10章运行计算图之间分布式的同步更新模式样例程序错误

### ISSUE THREAD

**Issue by MichaelXucf** [caicloud/tensorflow-tutorial#82]
你好，
     我在运行第10章"同步更新模式样例程序.py"时，最后训练完成后会报如下的错误：

> ERROR:tensorflow:==================================
Object was never used (type <class 'tensorflow.python.framework.ops.Tensor'>):
<tf.Tensor 'report_uninitialized_variables/boolean_mask/Gather:0' shape=(?,) dtype=string>
If you want to mark it as used call its "mark_used()" method.
It was originally created here:
['File "distributed_mnist_sync.py", line 177, in <module>\n    tf.app.run()', 'File "/usr/local/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 48, in run\n    _sys.exit(main(_sys.argv[:1] + flags_passthrough))', 'File "distributed_mnist_sync.py", line 109, in main\n    global_step, loss, train_op, opt = build_model(x, y_, n_workers, is_chief)', 'File "distributed_mnist_sync.py", line 69, in build_model\n    train_op = opt.minimize(loss, global_step=global_step)', 'File "/usr/local/lib/python2.7/site-packages/tensorflow/python/training/optimizer.py", line 351, in minimize\n    name=name)', 'File "/usr/local/lib/python2.7/site-packages/tensorflow/python/training/sync_replicas_optimizer.py", line 257, in apply_gradients\n    variables.global_variables())', 'File "/usr/local/lib/python2.7/site-packages/tensorflow/python/util/tf_should_use.py", line 175, in wrapped\n    return _add_should_use_warning(fn(*args, **kwargs))', 'File "/usr/local/lib/python2.7/site-packages/tensorflow/python/util/tf_should_use.py", line 144, in _add_should_use_warning\n    wrapped = TFShouldUseWarningWrapper(x)', 'File "/usr/local/lib/python2.7/site-packages/tensorflow/python/util/tf_should_use.py", line 101, in __init__\n    stack = [s.strip() for s in traceback.format_stack()]']

     我的运行环境：
     python 2.7.3
     tensorflow 1.3.0

  运行异步更新模型的例子是没有问题的， 运行同步更新就会有。
  这个问题不知道大家是否也有遇到过这种问题。
 在tensorflow的github上也有人提这个问题，希望帮助解决一下
https://github.com/tensorflow/tensorflow/issues/11823

**Comment by perhapszzy** [comment]
目前还没有支持tf 1.3，之后我们会将更新版本的代码更新。
